src/main/java/com/box/sdk/BoxWebHook.java

```diff
@@ -0,0 +1,685 @@
+         */
+        public TargetType getType() {
+            return this.type;
+        }
+
+        /**
+         * @return {@link BoxResource#getID()}
+         */
+        public String getId() {
+            return this.id;
+        }
+
+    }
+
+    /**
+     * Supported {@link Target} types.
+     *
+     * @author Stanislav Dvorscak
+     *
+     */
+    public enum TargetType {
+
+        /**
+         * WebHook for {@link BoxFolder} target type.
+         */
+        FOLDER("folder"),
+
+        /**
+         * WebHook for {@link BoxFile} target type.
+         */
+        FILE("file");
+
+        /**
+         * @see #getValue()
+         */
+        private final String value;
+
+        /**
+         * Constructor.
+         *
+         * @param value
+         *            {@link #getValue()}
+         */
+        TargetType(String value) {
+            this.value = value;
+        }
+
+        /**
+         * @return String representation for {@link TargetType}.
+         */
+        public String getValue() {
+            return this.value;
+        }
+
+        /**
+         * @param value value to get the TargetType enum value for
+         * @return TargetType for given value
+         */
+        public static TargetType fromValue(String value) {
+            for (TargetType targetType : TargetType.values()) {
+                if (targetType.getValue().equals(value)) {
+                    return targetType;
+                }
+            }
+            throw new IllegalArgumentException("No TargetType for value: " + value);
+        }
+    }
+
+    /**
+     * A Box related triggers.
+     *
+     * @author Stanislav Dvorscak
+     *
+     */
+    public enum Trigger {
+
+        // BoxFolder related triggers.
+
+        /**
+         * Triggered when a {@link BoxFolder} gets created.
+         */
+        FOLDER_CREATED("FOLDER.CREATED", TargetType.FOLDER),
+
+        /**
+         * Triggered when a {@link BoxFolder} gets copied.
+         */
+        FOLDER_COPIED("FOLDER.COPIED", TargetType.FOLDER),
+
+        /**
+         * Triggered when a {@link BoxFolder} gets moved.
+         */
+        FOLDER_MOVED("FOLDER.MOVED", TargetType.FOLDER),
+
+        /**
+         * Triggered when a {@link BoxFolder} is downloaded.
+         */
+        FOLDER_DOWNLOADED("FOLDER.DOWNLOADED", TargetType.FOLDER),
+
+        /**
+         * Triggered when a {@link BoxFolder} gets restored.
+         */
+        FOLDER_RESTORED("FOLDER.RESTORED", TargetType.FOLDER),
+
+        /**
+         * Triggered when a {@link BoxFolder} gets deleted.
+         */
+        FOLDER_DELETED("FOLDER.DELETED", TargetType.FOLDER),
+
+        // BoxFile related triggers.
+
+        /**
+         * Triggered when a {@link BoxFile} gets uploaded.
+         */
+        FILE_UPLOADED("FILE.UPLOADED", TargetType.FILE),
+
+        /**
+         * Triggered when a {@link BoxFile} gets copied.
+         */
+        FILE_COPIED("FILE.COPIED", TargetType.FILE),
```

Comment:
Other FILE.* and COMMENT.* triggers should be available on both target type file and folder.